nsg/vars.tf

```diff
@@ -24,4 +24,31 @@ variable "runner_subnet_address_prefixes" {
 variable "github_org_id" {
   description = "GitHub Organization ID"
   type        = string
-}
\ No newline at end of file
+}
+
+variable "include_log_analytics" {
+  description = "Create resources related to log analytics"
+  type = bool
+  default = false
+}
+
+variable "logging_storage_account_name" {
```

Comment:
This isn't backwards compatible, as it requires a `logging_storage_account_name` variable, regardless of the value of `include_log_analytics.`.  Azure storage account naming constraints require storage account  names to be unique across all of Azure, so to avoid this, we would need to either 1) programmatically generate a unique name when `include_log_analytics` is true, or 2) require the user to manage the storage account (and log analytics account) outside of the module and pass the id of those resources to the module for configuring the flow log.  I personally prefer option 2 so devs will have full access to configure those resources as desired, and will support this functionality when used against pre-existing storage accounts.  Since the network watcher requires multiple log analytics workspace (LAW) attributes, it might be better to pass in a name of a LAW and use a data resource to access those properties.




```
│ Error: Missing required argument
│ 
│   on main.tf line 1, in module "github_runner_vnet":
│    1: module "github_runner_vnet" {
│ 
│ The argument "logging_storage_account_name" is required, but no definition was found.
```

Reply:
OK. There's no reason for the changes for the ARM templates and these logging resources to be tied together, so I'll update my fork to remove the logging stuff for now so we can get the ARM template changes merged in. We can continue to spin on the logging stuff in another PR.